src/lib/components/selectv2/Selectv2.component.tsx

```diff
@@ -60,7 +60,10 @@ const InternalOption = (props) => {
   const formatOptionLabel = () => {
     const label = props.data.label;
     const inputValue = props.selectProps.inputValue;
-    const parts = label.split(new RegExp(`(${inputValue})`, 'gi'));
+    const parts = label.split(inputValue).flatMap((item, index) => {
+      if (index % 2 === 1) return [inputValue, item]
+      return [item]
+    });
```

Comment:
I don't think that this is correct.
For the following input: 'test-bla-test-bla', searching for test 
We expect `['', 'test', '-bla-', 'test', '-bla']` whereas with your implementation we get: `['', '-bla-', 'test', '-bla']`

Maybe simply do a `.flatMap((item) => [item, inputValue])` and remove the last entry

Reply:
I think it's correct, but I can do something simpler. :)

Reply:
Updated. :)